src/analysis/souffle/BUILD

```diff
@@ -103,6 +103,27 @@ souffle_cc_binary(
     src = ":sql_verifier_tag_unit_test_for_test",
 )
 
+gen_souffle_cxx_code(
+    name = "sql_verifier_code",
```

Comment:
Shouldn't we use this? https://github.com/google-research/raksha/blob/81f0d550c8d51944dc4f2551aba21a43d963d92d/src/backends/policy_engine/souffle/BUILD#L105

Or, consolidate them in one place.

Reply:
Consolidated in `sql_output.dl` due to the entangledness of Souffle datalog files.

Reply:
My suggestion was merely to consolidate the BUILD ruIes and the additional dl files in one place (wherever that is).  What we shouldn't be doing is forking this again. We already have put together some rules in https://github.com/google-research/raksha/blob/main/src/backends/policy_engine/souffle/BUILD#L90-L122  

If you think `src/analysis/souffle` is the right place, let us move them here and consolidate the analysis into one library.  The [souffle_policy_checker](https://github.com/google-research/raksha/blob/c00ce0a76a8632fe18160660868a4f702f1d2dc7/src/backends/policy_engine/souffle/BUILD#L125) rule depends on [datalog_policy_engine](https://github.com/google-research/raksha/blob/c00ce0a76a8632fe18160660868a4f702f1d2dc7/src/backends/policy_engine/souffle/BUILD#L105). Given that all your current changes are in a new rule, it won't be available in the souffle_policy_checker.   Let us consolidate the changes in one place and have `[souffle_policy_checker](https://github.com/google-research/raksha/blob/c00ce0a76a8632fe18160660868a4f702f1d2dc7/src/backends/policy_engine/souffle/BUILD#L125) and `sql_verifier_unit_test` depend on the same rule. 

Even though we are building sql specific code, let us still call it something generic like `datalog_policy_engine` as that is what we want it to be eventually.

Reply:
Consolidated dl files and rules, also renamed library target so this is `datalog_policy_verifier_cxx`, similar to suggested.